torch/_functorch/aot_autograd.py

```diff
@@ -451,6 +451,9 @@ def convert(idx, x):
                 if shape_env is not None:
                     from torch._dynamo.source import ConstantSource
                     if isinstance(x, int):
+                        # We always specialize on scalar values in export.
+                        if aot_config.is_export:
```

Comment:
Does export want to do this unconditionally? Or do we (or in the future want to) provide a way for export users to mark scalar inputs as dynamic

Reply:
We want to do it eventually (maybe next half?)